File: API/Entities/Ingredient.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Entities;
[Table("Ingredients")]
public class Ingredient
{
    public int Id { get; set; }
    public required string Name { get; set; }
    public int CategoryId { get; set; }
    public IngredientCategory Category { get; set; } = null!;
    public bool IsAllergen { get; set; } = false;
    public int? AllergyId { get; set; }
    public Allergy? Allergy { get; set; }
    public IngredientMeasurementType MeasurementType { get; set; } = IngredientMeasurementType.None;
    public float Calories { get; set; }
    public Photo? Photo { get; set; }

    public ICollection<RecipeIngredient> RecipeIngredients { get; set; } = [];
}

File: client/src/app/_models/ingredient.ts
import { Allergy } from "./allergy";
import { IngredientCategory } from "./ingredientCategory";
import { MeasurementType } from "./measurementType";
import { Photo } from "./photo";

export interface Ingredient {
  id : number;
  name : string;
  category : IngredientCategory;
  isAllergen : boolean;
  allergy : Allergy
  MeasurementType : MeasurementType;
  calories : number;
  photoUrl : string;
}
